### PR TITLE
WikiaPageType::isWikiaHubMain - allow a "custom" title to be passed

### PR DESCRIPTION
**Pull request by macbre**
[SUS-11](https://wikia-inc.atlassian.net/browse/SUS-11)

`WikiaHubsV3Hooks::onArticleFromTitle` calls `WikiaPageType::isWikiaHubMain` in the context where `$wgTitle` is not defined (namely `updateSpecialPages` script and tasks).

Allow `WikiaPageType::isWikiaHubMain` to be provided with a "custom" `Title` object to perform a check for (instead of relying on `$wgTitle`). Unit tests extended to cover this case.

@Wikia/sustaining-team

### CODE REVIEW

File: extensions/wikia/WikiaHubsV3/hooks/WikiaHubsV3Hooks.php [hunk 1]
```diff
@@ -7,11 +7,11 @@ class WikiaHubsV3Hooks {
 	 *
 	 * @return true because it's a hook
 	 */
-	static public function onArticleFromTitle(&$title, &$article) {
+	static public function onArticleFromTitle(Title $title, &$article) {
```
Comment (adamkarminski) [inline extensions/wikia/WikiaHubsV3/hooks/WikiaHubsV3Hooks.php:10]: If the `$article` is an object you could remove the reference and hint a class for it. :)

**Comment by adamkarminski** [comment]
Ultra mini minor comment, LGTM otherwise. :)

File: extensions/wikia/WikiaHubsV3/hooks/WikiaHubsV3Hooks.php (hunk 1 above)
Comment (macbre) [inline extensions/wikia/WikiaHubsV3/hooks/WikiaHubsV3Hooks.php:10]: Removed in 851fb45

Comment (macbre) [inline extensions/wikia/WikiaHubsV3/hooks/WikiaHubsV3Hooks.php:10]: The `$article` object is replaced inside the hook - let's keep the reference.

**Pull request merged**